File: madameautomata-bot-python-autoreg/bot/commands/domprofile.py

import discord
from discord import app_commands, Interaction
from ..theme import gothic_embed
from ..util.perm import is_domme, is_staff, is_sub
from ..db import Session, User, Collaring, Cage, JailSession, Star, Task, Worship, Config, CollarType, StarStatus
from ..util.time import parse_duration, random_case
from ..util.ids import cuid
from datetime import datetime, timedelta
from sqlalchemy import select, func, update, and_

def setup(bot):
    @bot.tree.command(name="domprofile", description="Show or set a Domme profile.")
    @app_commands.describe(titles="Preferred titles", about="About me", limits="Limits")
    async def domprofile(i: Interaction, titles: str | None = None, about: str | None = None, limits: str | None = None):
        async with Session() as s:
            me = await s.get(User, i.user.id)
            if titles or about or limits:
                if not me:
                    me = User(id=i.user.id, is_domme=True)
                    s.add(me)
                me.is_domme = True
                if titles is not None: me.titles = titles
                if about is not None: me.about = about
                if limits is not None: me.limits = limits
                await s.commit()
                return await i.response.send_message("Profile updated.", ephemeral=True)

            # read
            if not me:
                return await i.response.send_message("No profile yet. Add one with options.", ephemeral=True)
            # count active collars as domme
            res = await s.execute(select(Collaring).where(Collaring.domme_id==i.user.id, Collaring.active==True))
            count = len(res.scalars().all())
        e = gothic_embed("👑 Domme Profile", f"**Titles:** {me.titles or '—'}\n**About:** {me.about or '—'}\n**Limits:** {me.limits or '—'}\n**Collared currently:** {count}")
        await i.response.send_message(embed=e)

File: madameautomata-js-bot/src/commands/domprofile.js

import { SlashCommandBuilder, PermissionFlagsBits, ChannelType, ActionRowBuilder, ButtonBuilder, ButtonStyle, StringSelectMenuBuilder, EmbedBuilder } from "discord.js";
import { prisma } from "../lib/prisma.js";
import { Theme, gothicEmbed } from "../theme/theme.js";
import { parseDuration, randomCase } from "../util/duration.js";
import { cuid } from "../util/ids.js";
import { isDomme, isSub, isStaff, getConfig } from "../util/perm.js";

const cmd = 
{
  data: new SlashCommandBuilder()
    .setName("domprofile")
    .setDescription("Show or set a Domme profile.")
    .addStringOption(o=>o.setName("titles").setDescription("Preferred titles"))
    .addStringOption(o=>o.setName("about").setDescription("About me"))
    .addStringOption(o=>o.setName("limits").setDescription("Limits"))
    .addUserOption(o=>o.setName("user").setDescription("View another Domme")),
  async run(i) {
    const titles = i.options.getString("titles");
    const about = i.options.getString("about");
    const limits = i.options.getString("limits");
    const target = i.options.getUser("user") ?? i.user;
    if (target.id === i.user.id && (titles!==null || about!==null || limits!==null)) {
      await prisma.user.upsert({
        where: { id: i.user.id },
        update: { isDomme: true, ...(titles!==null?{titles}:{}) , ...(about!==null?{about}:{}) , ...(limits!==null?{limits}:{}) },
        create: { id: i.user.id, isDomme: true, titles, about, limits }
      });
      return i.reply({ content: "Profile updated.", ephemeral: true });
    }
    const tu = await prisma.user.findUnique({ where: { id: target.id } });
    const collared = await prisma.collaring.count({ where: { dommeId: target.id, active: true } });
    const e = gothicEmbed("👑 Domme Profile",
      `**User:** <@${target.id}>\n**Titles:** ${tu?.titles ?? "—"}\n**About:** ${tu?.about ?? "—"}\n**Limits:** ${tu?.limits ?? "—"}\n**Currently collared:** ${collared}`
    );
    return i.reply({ embeds: [e] });
  }
}


export default cmd;
